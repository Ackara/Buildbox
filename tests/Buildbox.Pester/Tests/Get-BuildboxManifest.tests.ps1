Import-Module "$(Split-Path $PSScriptRoot -Parent)\config.psm1" -Force
$context = New-TestEnvironment;
Import-Module $context.ModulePath -Force;

Describe "Get-BuildboxManifest" {
	Push-Location $context.TestDir;
	Context "Serialization" {
		$sampleFile = Get-Item "$($context.TestDataDir)\manifest.json";
		It "should deserialize a valid json file." {
			$result = Get-BuildboxManifest $sampleFile | ConvertTo-Json;
			{ Approve-Text $result "get-manifest1.json" } | Should Not Throw;
		}

		It "should be able to invoke [Manifest] methods" {
			$manifest = $sampleFile | Get-BuildboxManifest;
			$manifest.GetVersionSuffix("*") | Should Be "alpha";
		}

		It "should not overwrite the extended properties in the json file" {
			$sampleFile = "$($context.TestDataDir)\extended_manifest1.json";
			$manifest = Get-BuildboxManifest $sampleFile;
			$manifest.Id = "This field was modified";
			$manifest.Description = $null;
			$manifest.Save();

			{ Approve-File $sampleFile } | Should Not Throw;
		}

		It "should throw an exception when the specified file do not exist" {
			{ Get-BuildboxManifest -p "$($context.TestDir)\.nonexist.json" } | Should Throw;
		}
	}

	Pop-Location;
}