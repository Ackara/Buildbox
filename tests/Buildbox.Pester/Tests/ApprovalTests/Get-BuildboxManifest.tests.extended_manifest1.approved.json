{
    "id":  "This field was modified",
    "productName":  "Buildbox",
    "authors":  "Ackara",
    "owner":  "Ackara",
    "copyright":  "All rights reserved.",
    "projectUrl":  "https://example.com/project/buildbox",
    "iconUri":  "https://example.com/icon/buildbox.png",
    "licenseUri":  "https://example.com/buildbox/license.txt",
    "tags":  "build, deploy",
    "releaseNotes":  "some notes",
    "repositoryUrl":  "https://github.com/Ackara",
    "version":  {
                    "Major":  1,
                    "Minor":  2,
                    "Patch":  3,
                    "Suffix":  ""
                },
    "branchSuffixMap":  {
                            "*":  "alpha",
                            "dev":  "preview",
                            "master":  ""
                        },
    "feeds":  {
                  "nuget":  "nuget url here",
                  "vsts":  "vsts url here"
              }
}
